src/frontend/extract-helper.ts

```diff
@@ -0,0 +1,51 @@
+import { fetchS3 } from "../shared/utils";
+import { MONTHS } from "./frontend-utils";
+
+export const getContractPeriods = async (
+  contractId: string
+): Promise<Array<{ month: string; year: string; prettyMonth: string }>> => {
+  const dashboardData = await getDashboardExtract();
+  return dashboardData
+    .filter((row) => row.contract_id === contractId)
+    .map((row) => ({
+      year: row.year,
+      month: row.month,
+      prettyMonth: MONTHS[Number(row.month) - 1],
+    }))
+    .filter(
+      (row, i, rows) =>
+        rows.findIndex((r) => r.prettyMonth === row.prettyMonth) === i
+    ) // removes duplicates
+    .sort((a, b) => {
+      if (a.year === b.year) {
+        return a.month - b.month;
+      } else {
+        return a.year - b.year;
+      }
+    });
+};
+
+export const getLineItems = async (
```

Comment:
`getContractPeriods` and `getDashboardExtract` have unit tests but `getLineItems` doesn't.

Reply:
👍 .  The `getDashboardExtract` function doesn't really need to be exported as it's only used within this file, so now there are just `getContractPeriods` and `getLineItems` tests.